frontend/packages/operator-lifecycle-manager/src/components/descriptors/utils.tsx

```diff
@@ -134,54 +138,47 @@ const getCompatibleCapabilities = (type: string): (StatusCapability | SpecCapabi
 export function getValidCapabilitiesForDataType<CapabilityType extends string = SpecCapability>(
   descriptor: Descriptor<CapabilityType>,
   type: string,
-  allowDeprecated?: boolean,
 ): CapabilityType[] {
   const compatibleCapabilities = getCompatibleCapabilities(type);
-  const [valid, invalid, deprecated] = _.reduce(
-    descriptor?.['x-descriptors'] ?? [],
-    ([validAccumulator, invalidAccumulator, deprecatedAccumulator], capability) => {
-      const isDeprecated = DEPRECATED_CAPABILITIES.some((deprecatedCapability) =>
-        capability.startsWith(deprecatedCapability),
-      );
-
+  return (descriptor?.['x-descriptors'] ?? [])
+    .filter((capability) => {
       const isCompatible =
         type === 'any' ||
-        compatibleCapabilities.some((compatibleCapability) =>
+        compatibleCapabilities.findIndex((compatibleCapability) =>
```

Comment:
Why switch from `.some` which returns a boolean to `.findIndex` which returns in integer?

Reply:
This was an error, thanks for catching.